### PR TITLE
[#407] Add service tree client prototype

### LINKED ISSUE #407: Prototype of tree service from NeoFS Node
Tree service is a service of NeoFS Node that provides fast access to a node tree structure per container. This tree structure will store object related data, but we can use it for bucket configuration as well.

Consider moving some bucket configuration such as opt-in variables to the container tree (such as enabling versions, etc.). Tree root should be append with a node of bucket configuration `K:V` pairs. 

See https://github.com/fyrchik/neofs-node/blob/tree-service/pkg/services/tree/service.proto and `tree-service` branch in general.

Open PR into [tree-service](https://github.com/nspcc-dev/neofs-s3-gw/tree/tree-service) branch of S3 Gateway.

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #407 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: internal/neofs/tree.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	meta map[string]string
+}
+
+const (
+	versioningEnabledKV = "versioning_enabled"
+	lockConfigurationKV = "lock_configuration"
+	fileNameKV          = "FileName"
+	systemNameKV        = "SystemName"
+
+	settingsFileName = ".bucket-settings"
```
Comment (alexvanin) [inline internal/neofs/tree.go:34]: What you think about removing `.`?
/cc @masterSplinter01 @KirillovDenis

File: internal/neofs/tree.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+	result := make([]*tree.KeyValue, 0, len(meta))
+
+	for key, value := range meta {
+		result = append(result, &tree.KeyValue{Key: key, Value: []byte(value)})
+	}
+
+	return result
+}
+
+func parseLockConfiguration(value string) (*data.ObjectLockConfiguration, error) {
```
Comment (alexvanin) [inline internal/neofs/tree.go:225]: It seems that we have some strict encoding rules. Let's add some tests to stress that it is important.

**Review by alexvanin — APPROVED** [review]
LGTM. I would like to sync `tree-service` with `master` branch and merge it after. Thoughts?

File: go.mod [hunk 3]
```diff
@@ -5,23 +5,24 @@ go 1.16
...
 	github.com/bluele/gcache v0.0.2
-	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
 	github.com/golang/mock v1.6.0 // indirect
 	github.com/google/uuid v1.2.0
 	github.com/gorilla/mux v1.8.0
-	github.com/nats-io/nats-server/v2 v2.7.1 // indirect
-	github.com/nats-io/nats.go v1.13.1-0.20220121202836-972a071d373d
-	github.com/nspcc-dev/neo-go v0.98.0
+	github.com/nats-io/nats.go v1.13.1-0.20220308171302-2f2f6968e98d
+	github.com/nspcc-dev/neo-go v0.98.3-pre.0.20220421162730-3463d7292fb9
```
[the remark this answers is not in the document]
Comment (KirillovDenis) [reply go.mod:12]: We cannot do this until neofs-node also supports updated sdk

**Review by alexvanin — APPROVED** [review]

**Pull request merged**